examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+    else:
+        try:
+            sktime_flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
+            serialization_format = sktime_flavor_conf.get(
+                "serialization_format", SERIALIZATION_FORMAT_PICKLE
+            )
+        except MlflowException:
+            _logger.warning(
+                "Could not find sktime flavor configuration during model "
+                "loading process. Assuming 'pickle' serialization format."
+            )
+            serialization_format = SERIALIZATION_FORMAT_PICKLE
+
+        pyfunc_flavor_conf = _get_flavor_configuration(
+            model_path=path, flavor_name=pyfunc.FLAVOR_NAME
+        )
+        path = os.path.join(path, pyfunc_flavor_conf["model_path"])
+
+    return _SktimeModelWrapper(_load_model(path, serialization_format=serialization_format))
+
+
+class _SktimeModelWrapper:
+    def __init__(self, sktime_model):
+        self.sktime_model = sktime_model
+
+    def predict(self, X):
+        X = None if X.empty else X
+        raw_predictions = {}
+
+        if not hasattr(self.sktime_model, "pyfunc_predict_conf"):
+            raw_predictions[SKTIME_PREDICT] = self.sktime_model.predict(X=X)
+
+        else:
+            if not isinstance(self.sktime_model.pyfunc_predict_conf, dict):
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must be of type dict.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if PYFUNC_PREDICT_CONF_KEY not in self.sktime_model.pyfunc_predict_conf:
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must contain "
+                    f"a dictionary key {PYFUNC_PREDICT_CONF_KEY}.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], list):
+                predict_methods = self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY]
+                predict_params = False
+            elif isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], dict):
+                predict_methods = list(
+                    self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY].keys()
+                )
+                predict_params = True
+            else:
+                raise MlflowException(
+                    "Dictionary value must be of type dict or list.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if not set(predict_methods).issubset(set(SUPPORTED_SKTIME_PREDICT_METHODS)):
+                raise MlflowException(
+                    f"The provided {PYFUNC_PREDICT_CONF_KEY} values must be "
+                    f"a subset of {SUPPORTED_SKTIME_PREDICT_METHODS}",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if SKTIME_PREDICT in predict_methods:
```

Comment:
this should just be its own method in this class

Reply:
Each of these comments about creating methods is intended to increase readability of this class. 
`def predict()`, the implementation of the abstract method from the pyfunc wrapper, can handle simple orchestration of configuration which will call the discrete private methods that are encapsulated in this difficult to read if/elif/else series of decision chains.

Reply:
makes sense

Reply:
not relevant anymore with revised interface